addons/mass_mailing/models/mailing.py

```diff
@@ -463,6 +465,55 @@ def _group_expand_states(self, states, domain, order):
     # ACTIONS
     # ------------------------------------------------------
 
+    def action_add_favorite(self):
+        """Add the current mailing in the favorites list."""
+        self.ensure_one()
+
+        if self.is_body_empty:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'warning',
+                    'message': _('Please create a template before adding in your favorites list.'),
+                    'sticky': False,
+                    'next': {'type': 'ir.actions.act_window_close'},
+                }
+            }
+
+        self.write({
+            'favorite': True,
+            'favorite_date': fields.Datetime.now(),
+        })
+        return {
+            'type': 'ir.actions.client',
+            'tag': 'display_notification',
+            'params': {
+                'type': 'info',
+                'message': _('Design added to your templates'),
+                'sticky': False,
+                'next': {'type': 'ir.actions.act_window_close'},
+            }
+        }
+
+    def action_remove_favorite(self):
+        """Remove the current mailing from the favorites list."""
+        self.ensure_one()
```

Comment:
Same here, could be multi easily + computed field update that allows to avoid business logic a bit hidden in an action method ?